catroid/src/androidTest/java/org/catrobat/catroid/uiespresso/content/brick/app/VIsualPlacementActivityTest.java

```diff
@@ -0,0 +1,132 @@
+
+import org.catrobat.catroid.R;
+import org.catrobat.catroid.content.bricks.PlaceAtBrick;
+import org.catrobat.catroid.test.utils.TestUtils;
+import org.catrobat.catroid.testsuites.annotations.Cat;
+import org.catrobat.catroid.testsuites.annotations.Level;
+import org.catrobat.catroid.ui.SpriteActivity;
+import org.catrobat.catroid.uiespresso.util.rules.FragmentActivityTestRule;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+
+import static android.support.test.espresso.Espresso.onView;
+import static android.support.test.espresso.action.ViewActions.click;
+import static android.support.test.espresso.assertion.ViewAssertions.matches;
+import static android.support.test.espresso.matcher.ViewMatchers.isDisplayed;
+import static android.support.test.espresso.matcher.ViewMatchers.withId;
+import static android.support.test.espresso.matcher.ViewMatchers.withText;
+
+import static org.catrobat.catroid.uiespresso.content.brick.utils.BrickDataInteractionWrapper.onBrickAtPosition;
+import static org.catrobat.catroid.uiespresso.content.brick.utils.BrickTestUtils.createProjectAndGetStartScript;
+import static org.hamcrest.Matchers.equalToIgnoringWhiteSpace;
+
+@RunWith(AndroidJUnit4.class)
+public class VIsualPlacementActivityTest {
+	private int brickPosition;
+	private int xPos = 200;
+	private int yPos = -500;
+
+	@Rule
+	public FragmentActivityTestRule<SpriteActivity> baseActivityTestRule = new
+			FragmentActivityTestRule<>(SpriteActivity.class, SpriteActivity.EXTRA_FRAGMENT_POSITION, SpriteActivity.FRAGMENT_SCRIPTS);
+
+	@After
+	public void tearDown() throws Exception {
+		TestUtils.deleteProjects(VIsualPlacementActivityTest.class.getSimpleName());
+	}
+
+	@Before
+	public void setUp() throws Exception {
+		brickPosition = 1;
+		PlaceAtBrick brick = new PlaceAtBrick();
+		brick.setCoordinates(xPos, yPos);
+		createProjectAndGetStartScript(PlaceAtBrickTest.class.getSimpleName())
+				.addBrick(brick);
+		baseActivityTestRule.launchActivity();
+	}
+	@Category({Cat.AppUi.class, Level.Smoke.class})
+	@Test
+	public void testNoChangeConfirm() {
+
+		onBrickAtPosition(brickPosition).performClick();
+		onView(withText(R.string.brick_place_at_option_place_visually))
+				.check(matches(isDisplayed()))
+				.perform(click());
+
+		onView(withId(R.id.confirm))
+				.perform(click());
+
+		onView(withId(R.id.brick_place_at_edit_text_x))
+				.check(matches(withText(equalToIgnoringWhiteSpace(Integer.toString(xPos).replace("-", "- ")))));
+		onView(withId(R.id.brick_place_at_edit_text_y))
+				.check(matches(withText(equalToIgnoringWhiteSpace(Integer.toString(yPos).replace("-", "- ")))));
+	}
+
+	@Test
+	public void testNoChangeBackSave() {
+		onBrickAtPosition(brickPosition).performClick();
+		onView(withText(R.string.brick_place_at_option_place_visually))
+				.check(matches(isDisplayed()))
```

Comment:
do we need to check here again if it's displayed, since it's already checked in the smoke test?